include/mbedtls/ecdsa.h

```diff
@@ -235,6 +235,27 @@ int mbedtls_ecdsa_write_signature_det( mbedtls_ecdsa_context *ctx,
 #endif /* MBEDTLS_DEPRECATED_REMOVED */
 #endif /* MBEDTLS_ECDSA_DETERMINISTIC */
 
+/**
+ * \brief           Convert a signature from ASN.1 to a raw concatenation
+ *                  of {r,s}
+ *
+ * \param sig       Signature to be converted
+ * \param ssize     Size of the passed buffer
+ * \param byte_len  Length of a single number of the signature
+ * \param buf       Buffer pointer
+ * \param slen      Size of the written signature
+ *
+ * \note            The size of the buffer \c ssize should be at least
+ *                  2*byte_len bytes long, otherwise this function will
+ *                  return an error.
```

Comment:
We need to document the assumptions on the size of `buf` here, too. Also, I am wondering if we want to declare and implement this function as acting in place. As it stands, it requires a fresh buffer for the modified signature, and the implementation even internally allocated MPI's on the heap. I think we should avoid that and modify the signature in place without any dynamic allocation.

Reply:
I'd rather avoid functions that must modify the data in place. I'm all for _allowing_ the input and output buffers to overlap if it's practical, but not for _mandating_ it.

Reply:
Done - please re-review.